### PR TITLE
Feature/hand listener

### PR DESCRIPTION
**Pull request by davidpn11**
- Populates Player hands cards based on his session player hand
- Set currentCard
- Fix start deal hands bug

MISSSING:
- Add logic for non action cards on current Card.

### CODE REVIEW

File: src/api/db/preGameSession.ts [hunk 1]
```diff
@@ -241,10 +254,16 @@ export async function requestDealStartHands(session: LocalSessionWithId) {
...
   const buyCards = requestBuyCards(sessionRef);
 
-  const dealPlayerCard = (key: string, acc: Promise<SessionPlayerWithId>[], player: SessionPlayer) => {
-    return [...acc, buyCards({ id: key, ...player }, 7)];
-  };
+  const dealPlayerCard = async (
+    key: string,
+    acc: Promise<SessionPlayerWithId[]>,
+    player: SessionPlayer,
+  ) => [...(await acc), await buyCards({ id: key, ...player }, 7)];
```
Comment (mikelfcosta) [inline src/api/db/preGameSession.ts:261]: CRAZY

File: src/hooks/useHandListener.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { eqString } from 'fp-ts/lib/Eq';
+import { getPlayerHand } from '../store/playerHand/actions';
+
+export function useHandListener() {
+  const currentSession = useSelector(getSession);
+  const playerHand = useSelector(getPlayerHandIds);
+  const player = useSelector(getPlayer);
+  const [currPlayerHand, setCurrPlayerHand] = useState<string[]>([]);
+  const dispatch = useDispatch();
+  const E = A.getEq(eqString);
```
Comment (mikelfcosta) [inline src/hooks/useHandListener.tsx]: Give a name to this thing.

File: src/store/playerHand/actions.ts [hunk 3]
```diff
@@ -1,23 +1,51 @@
...
     type: SET_PLAYER_ID,
     payload: playerId,
   };
 }
 
-export type PlayerActionTypes = ReturnType<typeof setPlayer | typeof setPlayerId>;
+export function getPlayerHand(sessionId: string, hand: string[]) {
+  return async (dispatch: PlayerThunkDispatch) => {
+    try {
+      // console.log({ hand });
```
Comment (mikelfcosta) [inline src/store/playerHand/actions.ts:38]: comments

File: src/views/GameRouter/components/CardDeck/index.tsx [hunk 4]
```diff
@@ -12,8 +12,8 @@ const cards = Array(30).fill({
 export default function CardDeck() {
   return (
     <CardsWrapper>
-      {cards.map(card => (
-        <PlayingCard key={card.id} color={card.color} status="DECK" value={card.value} />
+      {cards.map((card, index) => (
+        <PlayingCard key={index} color={card.color} status="DECK" value={card.value} />
```
Comment (mikelfcosta) [inline src/views/GameRouter/components/CardDeck/index.tsx:16]: Cannot be index.

File: src/views/GameRouter/components/GameTable/index.tsx [hunk 5]
```diff
@@ -5,161 +5,31 @@ import { SessionPlayer } from '../../../../model/Player';
...
+  const renderPlayerCard = (key: string, player: SessionPlayer, pos: number) => (
+    <TablePlayer key={key} player={player} playerPosition={pos} />
+  );
+  const renderPlayers = () => {
+    let i = 0;
+    return pipe(
+      otherPlayers,
+      R.reduceWithIndex([] as JSX.Element[], (key, acc, p) => {
+        const player = renderPlayerCard(key, p, i);
+        i = i + 1;
```
Comment (mikelfcosta) [inline src/views/GameRouter/components/GameTable/index.tsx:28]: i += 1

**Review by mikelfcosta — APPROVED** [review]

File: src/views/GameRouter/components/CardDeck/index.tsx (hunk 4 above)
Comment (davidpn11) [reply src/views/GameRouter/components/CardDeck/index.tsx:16]: We will remove this

**Pull request merged**